### PR TITLE
[price-feeder] Add handling for jaield validator in oracle tick

### PR DESCRIPTION
**Pull request by udpatil**
## Describe your changes and provide context
This adds a check for whether the validator is jailed, and if so, doesn't allow submission of oracle votes, and instead logs error output

## Testing performed to validate your change
Verified with personal cluster

### CODE REVIEW

File: oracle/price-feeder/oracle/jail.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		return false
+	}
+
+	// This is an edge case, which should never happen.
+	// The current blockchain height is lower
+	// than the last updated block, to fix we should
+	// just update the cached params again.
+	if currentBlockHeight < jailCache.lastUpdatedBlock {
+		return true
+	}
```
Comment (BrandonWeng) [inline oracle/price-feeder/oracle/jail.go]: Hmm, if it should never happen then should this be a panic or at least log an error?

File: oracle/price-feeder/oracle/oracle.go [hunk 2]
```diff
@@ -417,6 +420,46 @@ func (o *Oracle) GetParams(ctx context.Context) (oracletypes.Params, error) {
...
+	if err != nil {
+		return false, err
+	}
+
+	o.jailCache.Update(currentBlockHeight, isJailed)
+	return isJailed, nil
+}
+
+// GetJailedState returns the current on-chain jailing state of the validator
+func (o *Oracle) GetJailedState(ctx context.Context) (bool, error) {
```
Comment (yzang2019) [inline oracle/price-feeder/oracle/oracle.go]: Can we move this to jail.go?

File: oracle/price-feeder/oracle/oracle.go [hunk 3]
```diff
@@ -417,6 +420,46 @@ func (o *Oracle) GetParams(ctx context.Context) (oracletypes.Params, error) {
 	return queryResponse.Params, nil
 }
 
+func (o *Oracle) GetJailedStateCache(ctx context.Context, currentBlockHeight int64) (bool, error) {
```
Comment (yzang2019) [inline oracle/price-feeder/oracle/oracle.go]: Let's move this to jail.go?

File: oracle/price-feeder/oracle/oracle.go [hunk 4]
```diff
@@ -417,6 +420,46 @@ func (o *Oracle) GetParams(ctx context.Context) (oracletypes.Params, error) {
...
+		o.oracleClient.GRPCEndpoint,
+		// the Cosmos SDK doesn't support any transport security mechanism
+		grpc.WithInsecure(),
+		grpc.WithContextDialer(dialerFunc),
+	)
+	if err != nil {
+		return false, fmt.Errorf("failed to dial Cosmos gRPC service: %w", err)
+	}
+
+	defer grpcConn.Close()
```
Comment (yzang2019) [inline oracle/price-feeder/oracle/oracle.go]: Can we keep the grpcConn open?

File: oracle/price-feeder/oracle/jail.go (hunk 1 above)
Comment (yzang2019) [reply oracle/price-feeder/oracle/jail.go]: Yeah, we don't really need this and the above check. It should only become stale if currentBlockHeight > lastUpdate + interval

**Comment by yzang2019** [comment]
Apart from adding block height based stale, we should probably add time based staleness check. This would cover the case where the node halted and become jailed, where the curr height doesn't move any more, but we should still stop sending any request right?

File: oracle/price-feeder/oracle/jail.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package oracle
+
+const (
+	jailCacheInterval = int64(50)
```
Comment (philipsu522) [inline oracle/price-feeder/oracle/jail.go]: nit - can you add units to the name of this var?

File: oracle/price-feeder/oracle/oracle.go (hunk 3 above)
Comment (philipsu522) [reply oracle/price-feeder/oracle/oracle.go]: fucntion seems to return jail state rather than the actual cache - should we rename?

Comment (philipsu522) [reply oracle/price-feeder/oracle/oracle.go]: or would `GetCachedJailState` make more sense? I thought we were returning some cache like
```

type JailCache struct {
	isJailed         bool
	lastUpdatedBlock int64
}
```

**Review by philipsu522 — APPROVED** [review]
more of a meta point, does it make sense to start having metrics here?

File: oracle/price-feeder/oracle/oracle.go (hunk 4 above)
Comment (udpatil) [reply oracle/price-feeder/oracle/oracle.go]: I followed the behavior here:
https://github.com/sei-protocol/sei-chain/blob/7f5681547fd498c5fbdd9fc8af45e200caae7e33/oracle/price-feeder/oracle/oracle.go#L408
where it closes the connection after the GetParams, I guess we could open one at the start of oracle and leave it open and then only close it when we terminate the price feeder?

**Pull request merged**